### ISSUE TITLE
Benchmark for CW17 stopped unexpectedly

### ISSUE THREAD

**Issue by pihme** [camunda-cloud/zeebe#7033]
**Describe the bug**
Benchmark for CW17 stopped unexpectedly: http://34.77.165.228/d/I4lo7_EZk/zeebe?orgId=1&var-DS_PROMETHEUS=Prometheus&var-namespace=medic-cw-17-df496d4b4-benchmark&var-pod=All&var-partition=All&from=1620720521670&to=1620725392887

![image](https://user-images.githubusercontent.com/14032870/117941478-6a560800-b30a-11eb-8373-a4f44db02b4c.png)

- [x] Resources look good
- [x] Snapshots are small (5 MB and few files ~9)

Issues
- none of the Zeebe broker pods are ready
![image](https://user-images.githubusercontent.com/14032870/117942194-17c91b80-b30b-11eb-8d1c-e2f80a61fd2b.png)

**Comment by pihme** [comment]
Repeated exception in logs is:
```
java.util.concurrent.CompletionException: java.lang.IllegalStateException: Expected to find a snapshot at index >= log's first index 1071151229, but found snapshot 0
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:314) ~[?:?]
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:319) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1081) ~[?:?]
	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:478) ~[?:?]
	at io.atomix.utils.concurrent.SingleThreadContext$WrappedRunnable.run(SingleThreadContext.java:188) ~[atomix-utils-1.0.0-SNAPSHOT.jar:1.0.0-SNAPSHOT]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
	at java.lang.Thread.run(Thread.java:829) ~[?:?]
Caused by: java.lang.IllegalStateException: Expected to find a snapshot at index >= log's first index 1071151229, but found snapshot 0
	at io.atomix.raft.impl.RaftContext.verifySnapshotLogConsistent(RaftContext.java:190) ~[atomix-cluster-1.0.0-SNAPSHOT.jar:1.0.0-SNAPSHOT]
	at io.atomix.raft.impl.RaftContext.<init>(RaftContext.java:166) ~[atomix-cluster-1.0.0-SNAPSHOT.jar:1.0.0-SNAPSHOT]
	at io.atomix.raft.impl.DefaultRaftServer$Builder.build(DefaultRaftServer.java:257) ~[atomix-cluster-1.0.0-SNAPSHOT.jar:1.0.0-SNAPSHOT]
	at io.atomix.raft.impl.DefaultRaftServer$Builder.build(DefaultRaftServer.java:231) ~[atomix-cluster-1.0.0-SNAPSHOT.jar:1.0.0-SNAPSHOT]
	at io.atomix.raft.partition.impl.RaftPartitionServer.buildServer(RaftPartitionServer.java:175) ~[atomix-cluster-1.0.0-SNAPSHOT.jar:1.0.0-SNAPSHOT]
	at io.atomix.raft.partition.impl.RaftPartitionServer.initServer(RaftPartitionServer.java:145) ~[atomix-cluster-1.0.0-SNAPSHOT.jar:1.0.0-SNAPSHOT]
	at io.atomix.raft.partition.impl.RaftPartitionServer.start(RaftPartitionServer.java:103) ~[atomix-cluster-1.0.0-SNAPSHOT.jar:1.0.0-SNAPSHOT]
	at io.atomix.raft.partition.RaftPartition.open(RaftPartition.java:113) ~[atomix-cluster-1.0.0-SNAPSHOT.jar:1.0.0-SNAPSHOT]
	at io.atomix.raft.partition.RaftPartitionGroup.lambda$join$6(RaftPartitionGroup.java:201) ~[atomix-cluster-1.0.0-SNAPSHOT.jar:1.0.0-SNAPSHOT]
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195) ~[?:?]
	at java.util.HashMap$KeySpliterator.forEachRemaining(HashMap.java:1603) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474) ~[?:?]
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578) ~[?:?]
	at io.atomix.raft.partition.RaftPartitionGroup.join(RaftPartitionGroup.java:203) ~[atomix-cluster-1.0.0-SNAPSHOT.jar:1.0.0-SNAPSHOT]
	at io.atomix.primitive.partition.impl.DefaultPartitionService.lambda$start$1(DefaultPartitionService.java:80) ~[atomix-cluster-1.0.0-SNAPSHOT.jar:1.0.0-SNAPSHOT]
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195) ~[?:?]
	at java.util.concurrent.ConcurrentHashMap$ValueSpliterator.forEachRemaining(ConcurrentHashMap.java:3605) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474) ~[?:?]
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578) ~[?:?]
	at io.atomix.primitive.partition.impl.DefaultPartitionService.start(DefaultPartitionService.java:81) ~[atomix-cluster-1.0.0-SNAPSHOT.jar:1.0.0-SNAPSHOT]
	at io.atomix.core.Atomix.lambda$startServices$0(Atomix.java:178) ~[atomix-1.0.0-SNAPSHOT.jar:1.0.0-SNAPSHOT]
	at java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1072) ~[?:?]
	... 8 more
```

**Comment by deepthidevaki** [comment]
This should be already fixed in 1.0.0 release. This happens because the Snapshot checksum file is not flushed to disk before the snapshot is marked as valid. Hence after a restart, it is not able to read the checksum file.

**Comment by pihme** [comment]
close as duplicate; fix was merged in CW18 so was not active in CW17 benchmark
